src/library/Statistics.nim

```diff
@@ -134,24 +134,25 @@ proc defineSymbols*() =
         example     = """
             print median [2 4 5 6 7 2 3]
             ; 6
-            
+
             print median [1 5 2 3 4 7 9 8]
             ; 3.5
         """:
             #=======================================================
-            if x.a.len==0: 
+            if x.a.len==0:
                 push(VNULL)
             else:
+                x.a = x.a.sorted(order = SortOrder.Ascending)
```

Comment:
```suggestion
                let sorted = x.a.sorted(order = SortOrder.Ascending)
```

...and then use `sorted` instead of `x.a`, would be a good idea.

Basically, there is a potential problem with directly sorting and setting `x.a` (haven't tried, you may; but I have a feeling it won't work properly). In a few words, I think the end result of this would be:

```
a: [3 1 2 4]
print median a ; it will print 2.5
print a ; now the array has been sorted, which was not intended after all
```

Not 100% sure, but I think it's quite probable. If I'm wrong, obviously you can ignore this comment 😉

Reply:
Just tested it, and unfortunately I was right. The `median` function now works properly (which is great), but the issue with the passed array being changed is there too:

<img width="204" alt="Screenshot 2023-02-20 at 11 09 40" src="https://user-images.githubusercontent.com/1265028/220076098-c669c1cf-89de-4d88-8477-946b505cf547.png">

Reply:
I'll solve it soon. At night I'll be home.
You're 100% alright here.
*Mea culpa!*

I missed this part, array should never be modified by this function.

Reply:
No worries! It's an easy trap, in which I've fallen more times than I can count! haha

It's just that those `x`, `y`, `z` that are auto-magically appear available in built-in function implementation are nothing but pointers. So, we may use them, but not change them. (or anything they point to...) 😉